src/components/pages/public/MovieDetailPage.vue

```diff
@@ -1,19 +1,89 @@
 <template>
-  <div>
-    Single movie of id:
-    {{ movieId }}
+  <div class="movie-detail-page">
+    <MovieDetailCard
+      :movieDetails="movieDetails"
+      class="movie-detail-page__movie-card"
+    />
+    <div>
+      <h2 class="movie-detail-page__screening-title">Screenings:</h2>
+      <p class="movie-detail-page__screening-content">
+        {{ selectedWeekday }} {{ formattedDate }}
+      </p>
+      <DayTabsList
+        class="movie-detail-page__day-tabs-list"
+        @set-date="setDate"
+      />
+      <ScreeningCard
+        class="movie-detail-page__screening-card"
+        :movie="movieDetails"
+        :filteredScreenings="filterByDay(movieId, selectedDate)"
+      />
+    </div>
   </div>
 </template>
 
 <script>
+import MovieDetailCard from "@/components/MovieDetailCard.vue";
+import DayTabsList from "@/components/DayTabsList.vue";
+import ScreeningCard from "@/components/ScreeningCard.vue";
+import dayTabsMixin from "@/mixins/dayTabs.js";
+import { getDayOfWeek, getDate, getTime } from "@/helpers";
+import { getMovieDetails } from "@/api/movies";
+import screeningsList from "@/mixins/screeningsList.js";
+
 export default {
+  name: "MovieDetailPage",
+  components: { MovieDetailCard, DayTabsList, ScreeningCard },
+  data() {
+    return {
+      movieDetails: {},
+    };
+  },
+  mixins: [dayTabsMixin, screeningsList],
+  methods: {
+    getDayOfWeek,
+    getDate,
+    getTime,
+  },
   props: {
     movieId: {
-      type: Number,
-      default: 0,
+      type: String,
+      required: true,
     },
   },
+  async created() {
+    const { data } = await getMovieDetails(this.movieId);
+    this.movieDetails = data;
+    this.$store.dispatch("setMovieTitle", this.movieDetails.title);
```

Comment:
I saw that you've used `mapGetters` earlier, you can do the same for actions. In your `methods` property you can add `...mapActions(<moduleName>, ['moduleMethod1', 'moduleMethod2'])`. Then in code you can use it just like any other method.